### PR TITLE
Add plain byte alphabet as a csa alphabet strategy

### PR DESCRIPTION
**Pull request by joergi-w**
We implemented the class `plain_byte_alphabet` and added tests and documentation.

Touches https://github.com/seqan/seqan3/issues/1603

### CODE REVIEW

File: include/sdsl/csa_alphabet_strategy.hpp [hunk 1]
```diff
@@ -542,6 +542,173 @@ void init_char_bitvector(sd_vector<t_hi_bit_vector, t_select_1, t_select_0> &cha
...
+        mapping_wrapper() = default;
+
+        //! Random access operator.
+        constexpr char_type operator[](char_type const c) const noexcept
+        {
+            return c;
+        }
+    };
+
+    const char2comp_type char2comp;
```
Comment (eseiler) [inline include/sdsl/csa_alphabet_strategy.hpp]:
```suggestion
    const char2comp_type char2comp{};
```

Fixes the clang-3.8 error

File: include/sdsl/csa_alphabet_strategy.hpp [hunk 2]
```diff
@@ -542,6 +542,173 @@ void init_char_bitvector(sd_vector<t_hi_bit_vector, t_select_1, t_select_0> &cha
...
+    const comp2char_type comp2char;
```
Comment (eseiler) [inline include/sdsl/csa_alphabet_strategy.hpp]:
```suggestion
    const comp2char_type comp2char{};
```

File: include/sdsl/csa_alphabet_strategy.hpp [hunk 3]
```diff
@@ -198,7 +198,7 @@ class byte_alphabet {
         , m_sigma(bas.m_sigma)
     {
     }
-
```
Comment (eseiler) [inline include/sdsl/csa_alphabet_strategy.hpp]: remove

File: include/sdsl/csa_alphabet_strategy.hpp [hunk 4]
```diff
@@ -51,7 +51,7 @@ class byte_alphabet;
 template <class bit_vector_type   = bit_vector,
 		  class rank_support_type = rank_support_scan<>,
 		  class select_support_type =
-		  select_support_scan<>,
```
Comment (eseiler) [inline include/sdsl/csa_alphabet_strategy.hpp]: remove

File: include/sdsl/csa_alphabet_strategy.hpp (hunk 1 above)
Comment (joergi-w) [reply include/sdsl/csa_alphabet_strategy.hpp]: > Fixes the clang-3.8 error

Thanks!

**Pull request merged**